doc/cprover-manual/contracts-frees.md

```diff
@@ -0,0 +1,161 @@
+[CPROVER Manual TOC](../../)
+
+# Frees Clauses
+
+A _frees clause_ lets the user specify the set of pointers to dynamically
+allocated objects that may be freed by a function or a loop.
+
+## Syntax
+
+The clause has the following syntax:
+
+```c
+__CPROVER_frees( *targets* )
+```
+
+Where `targets` have the following syntax:
+
+```
+targets ::= conditional-target-group (';' conditional-target-group)* ';'?
+conditional-target-group ::= (condition ':')? target (',' target)*
+target ::= pointer-typed-expression
+        | user-defined-function(args)
+```
+
+Where `user-defined-function` is a user-defined function that must return
+the built-in type `__CPROVER_freeable_t` and be side-effect free.
+
+The condition expression may contain calls to side-effect-free functions.
+
+For function contracts, the target expressions may only involve function
+parameters, and variable or function identifiers defined in the global scope.
+
+For a loop contract, the target expressions may involve any identifier that is
+in scope at loop entry (parameters of the surrounding function, local variables,
+global variables, function identifiers, etc.).
+
+### Pointer-typed targets
+
+Pointer-typed targets can be directly listed in the clause.
+
+In a function contract:
+```c
+int foo(char *arr1, char *arr2, size_t size)
```

Comment:
Are these examples included as regression tests?  If so, perhaps there is some way to link to them.

Reply:
no these are just embedded in the doc.